cmd/microcloud/preseed.go

```diff
@@ -100,8 +107,40 @@ func DiskOperatorSet() filter.OperatorSet {
 	}
 }
 
+type cmdPreseed struct {
+	common *CmdControl
+
+	flagAdd bool
+}
+
+func (c *cmdPreseed) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "preseed",
+		Short: "Initialize and extend a MicroCloud cluster unattended",
+		RunE:  c.Run,
+	}
+
+	cmd.Flags().BoolVar(&c.flagAdd, "add", false, "Add new systems to an existing MicroCloud cluster")
```

Comment:
Can you not infer `add` without the flag if the following is true?
* `initiator` is set
* The `initiator` system is not in the `systems` list
* The local system is bootstrapped, and the `initiator` name matches its cluster name.

Reply:
Ah, I forgot about the unicast case when adding nodes which requires the joiners to know about the initiator's address. 

I guess you're currently handling this by adding an entry to the `systems` list including an initiator node that is already clustered.

I'm worried that might be a bit confusing because there are other keys that are valid for a `systems` entry. Either we ignore these keys (which would be confusing to the user), or we would have to add special-case validation for this entry so that the user can't specify an uplink network or storage configuration for the initiator when adding nodes.

Instead, I think it might be simpler to just add another key:
`initiator_address` that can be used instead of (or in addition to) `initiator`. If both are specified, then we must verify the address belongs to the node with that name.

Then we don't also need a `--add` flag, since the cases would look like this:
### init
* `initiator: m1`
* `initiator_address: 10.0.0.10` (no effect except validation with `initiator`)
* `systems: m1, m2, m3` (with addresses if unicast)

### add
* `initiator: m1`
* `initiator_address: 10.0.0.10` (required for unicast, no effect except validation with `initiator` for multicast)
* `systems: m2, m3` (with addresses if unicast)

Reply:
I like this idea.

If I don't forgot any scenario, we would not even need `initiator`. By having only `initiator_address`, we can check both scenarios too:

* **init**: Don't set `initiator_address`, pick the first one in the list of systems. If it's set, there has to be a system in the list matching this address.
* **add**: Imply we are in this mode if `initiator_address` is set and if it's not in the list of already specified system addresses.

Reply:
Actually you were right with having both the `initiator` and `initiator_address`. If there is just `initiator_address` you cannot grow the cluster using multicast as you would need to specify the address of an existing cluster member which again defeats the purpose of multicast for preseed.

By having `initiator` and `initiator_address` we can also get rid of the concept of picking the "first" system as we can enforce you specify either `initiator` (when doing multicast) or `initiator_address` (when doing unicast).

Reply:
Fixed.